docs/docker/building-net-docker-images.md

```diff
@@ -0,0 +1,189 @@
+
+## Docker image variatons
+
+The `microsoft/dotnet` images are offered in three different variations, each for a specific use case.
+
+- `microsoft/dotnet:<version>-sdk` : that is **microsoft/dotnet-preview2-sdk**, this image contains the .NET Core SDK which includes the .NET Core and Command Line Tools (CLI). Use this image for your development processes; dev, test debug.
+
+- `microsoft/dotnet:<version>-onbuild` : that is **microsoft/dotnet:onbuild**, contains `ONBUILD` triggers which should cover most applications. The build will `COPY` you application, run `dotnet restore` and create an `ENTRYPOINT dotnet run` instruction to run the application when the Docker image is run.
+
+- `microsoft/dotnet:<version>-core` : that is **microsoft/dotnet:1.0.0-core**, image contains only .NET Core (runtime and libraries) and it is optimized for running [portable .NET Core applications](https://docs.microsoft.com/en-us/dotnet/articles/core/app-types).
+
+- `microsoft/dotnet:<version>-core-deps` : that is **microsoft/dotnet:1.0.0-core-deps**, if you wish to run self-contained applications use this image.  Contains the operating system with all of the native dependencies needed bt .NET Core. Can also be used to build a custom copy of .NET core by compiling coreclr and corefx.
+
+Latest versions of each variant:
+
+- `microsoft/dotnet` or `microsoft/dotnet:latest` (includes SDK)
+- `microsoft/dotnet:onbuild`
+- `microsoft/dotnet:core`
+- `microsoft/dotnet:core-deps`
+
+Here is a list of the images after a `docker pull <imagename>` on a development machine to show the various sizes.
+
+```
+REPOSITORY          TAG                  IMAGE ID            SIZE
+microsoft/dotnet    onbuild              19b6a6c4b1db        540.4 MB
+microsoft/dotnet    1.0.0-preview2-sdk   a92c3d9ad0e7        540.4 MB
+microsoft/dotnet    core                 5224a9f2a2aa        253.2 MB
+microsoft/dotnet    1.0.0-core-deps      c981a2eebe0e        166.2 MB
+microsoft/dotnet    core-deps            c981a2eebe0e        166.2 MB
+microsoft/dotnet    latest               03c10abbd08a        540.4 MB
+microsoft/dotnet    1.0.0-core           b8da4a1fd280        253.2 MB
+```
+
+## Prerequisites
+
+- [.NET Core](http://dot.net)
+- Docker Engine : see [Docker Installation page](http://www.docker.com/products/docker)
+- Yeoman tooling for ASP.NET for creating the Web API application
+- Yeoman tools for Docker from Microsoft
+
+Install the Yeoman generators for ASP.NET Core and Docker using npm 
+
+```
+npm install -g yo generator-aspnet generator-docker
+```
+
+> [!NOTE]
+> This sample will be using [Visual Studio Code](http://code.visualstudio.com) for the editor.
+
+## Creating the Web API application
+
+Using the ASP.NET Yeoman generator, open a command or terminal session and type the following.
```

Comment:
nit: This sentence reads backwards to me: You don't use the Yeoman generator to open a terminal, you open a terminal and then type 'yo' to use the Yeoman generator.